src/tsickle.ts

```diff
@@ -105,89 +105,177 @@ class ClosureRewriter extends Rewriter {
     super(file);
   }
 
-  emitFunctionType(fnDecl: ts.SignatureDeclaration, extraTags: jsdoc.Tag[] = []) {
+  /**
+   * Handles emittng the jsdoc for methods, including overloads.
+   * If overloaded, merges the signatures in the list of SignatureDeclarations into a single jsdoc.
+   * - Total number of parameters will be the maximum count found across all variants.
+   * - Any parameters beyond the minimum count found across all variants; prefix: "opt_"
+   * - Different names at the same parameter index will be joined with "_or_"
+   * - Variable args (...type[] in TypeScript) will be output as "...type",
+   *    except if found at the same index as another argument.
+   */
+  emitFunctionType(fnDecls: ts.SignatureDeclaration[], extraTags: jsdoc.Tag[] = []): string[] {
     let typeChecker = this.program.getTypeChecker();
-    let sig = typeChecker.getSignatureFromDeclaration(fnDecl);
-
-    // Construct the JSDoc comment by reading the existing JSDoc, if
-    // any, and merging it with the known types of the function
-    // parameters and return type.
-    let jsDoc = this.getJSDoc(fnDecl) || [];
+    let minCount = 9999;
+    let maxCount = 0;
     let newDoc = extraTags;
-
-    // Copy all the tags other than @param/@return into the new
-    // comment without any change; @param/@return are handled later.
-    for (let tag of jsDoc) {
-      if (tag.tagName === 'param' || tag.tagName === 'return') continue;
-      newDoc.push(tag);
+    let paramNamesToReturn: string[] = [];
+    const abstract = {tagName: 'abstract'};
+    const isOverloaded = fnDecls.length > 1;
+    const isConstructor = fnDecls[0].kind === ts.SyntaxKind.Constructor;
+
+    for (let fnDecl of fnDecls) {
+      if (fnDecl.parameters.length < minCount) {
+        minCount = fnDecl.parameters.length;
+      }
+      if (fnDecl.parameters.length > maxCount) {
+        maxCount = fnDecl.parameters.length;
+      }
     }
+    let paramData = {
+      names: new Array<Array<string>>(maxCount),
+      types: new Array<Array<string>>(maxCount),
+      returns: ''
+    };
 
-    // Abstract
-    if ((fnDecl.flags & ts.NodeFlags.Abstract)) {
-      newDoc.push({tagName: 'abstract'});
-    }
+    for (let fnDecl of fnDecls) {
+      // Construct the JSDoc comment by reading the existing JSDoc, if
+      // any, and merging it with the known types of the function
+      // parameters and return type.
+      let jsDoc = this.getJSDoc(fnDecl) || [];
+
+      // Copy all the tags other than @param/@return into the new
+      // comment without any change; @param/@return are handled later.
```

Comment:
If there are overloads, does this mean we end up with multiple copies of the @foo listed on them?  E.g. if each overload has @see http://foo on them, we'll have multiple?

(I'm not sure there's any @foo we even care about preserving here though!)